csrc/cache.h

```diff
@@ -1,21 +1,22 @@
 #pragma once
 
-#include <torch/extension.h>
+#include <torch/all.h>
 
 #include <map>
 #include <vector>
 
 void swap_blocks(torch::Tensor& src, torch::Tensor& dst,
                  const torch::Tensor& block_mapping);
 
-void copy_blocks(std::vector<torch::Tensor>& key_caches,
-                 std::vector<torch::Tensor>& value_caches,
+void copy_blocks(std::vector<torch::Tensor> const& key_caches,
```

Comment:
Since the tensors inside the vector seem to be modified, would it be better to drop the `const` keyword?

Reply:
I had to add the `const`s here to satisfy pytorch's binding code.

Reply:
Can you add a comment about that (something along the lines that the tensors in here are modified but the const is needed for pytorch's binding code) :)

Reply:
@bnellnm Do you have any more details on why this is needed? Any error message or related issue you could share?

Reply:
There's just a limitation in pytorch's C++ library wrapper code.  I can paste a C++ template compilation error here if you want. :)

Reply:
OK, no worries - I can try it myself if I really get curious :)